### ISSUE TITLE
Gradle should be more tolerant of incorrectly formatted module metadata

### ISSUE THREAD

**Issue by jamulv** [gradle/gradle#27855]
### Current Behaviour

Gradle currently fails if module metadata is incorrect (not valid / valid json i.e. html):

```
Could not parse module metadata <url> 
    Use JsonReader.setLenient(true) to accept malformed JSON at <line>
```

Many maven repositories (artifactory / nexus) do not validate gradle .module files and will quite happily cache say an html page with invalid content. This recently had a knock on effect of taking nearly all our gradle builds offline as multiple repositories  outside of our organization cached invalid module files.
Effectively a denial of service attack on our builds

While we could change every build to use:
```
repositories {
  maven {
    metadataSources {
      mavenPom() //use maven pom
      artifact() //look directly for the artifact
      ignoreGradleMetadataRedirection() //it fixes the issue actually
    }
  }
}
```

This was too large an undertaking and caused issues with gradle plugins

### Expected Behavior

We would request that gradle tolerates invalid json in the module and logs this as a warning dropping down to .pom.

### Context (optional)

Large number of builds simultaneously impacted by bad .module files

### Steps to Reproduce

1. reference an artifact with a .module file containing html
2. gradle build fails as this is not json

### Gradle version

8.1.1

**Comment by Vampire** [comment]
This would be bad for build reproducibility though, for the same reason error codes besides 404 from a repository make the build fail and not look into other repositories.

If you have repositories A and B.
Then try to resolve artifact m.
If A says 404 for m, B is checked.
But if A gives a different error, resolution fails, as it could for example be a temporary failure, or changed credentials or something and just going to B could change the build outcome as B might have a different version of m.

The same would be true for discarding invalid GMM.
Today you get the proper GMM delivered, Gradle uses it.
Tomorrow you get an error besides 404 for the GMM and suddenly Gradle resolves things differently.

**Comment by jamulv** [comment]
When dealing with an upstream virtual repository that groups multiple repositories there is always a risk that as upstream repositories go up and down resolution can shift and change.
The issue in this specific case is as follows:
 - many artifacts declare a module but do not publish it (elastic client and many others)
 - we hit a mis-configured artifactory repo which returned an html page back as a response with 200 in response to the .module request
 - neither artifactory nor nexus have the logic to validate .module files so treat the content as verbatim (not the case for a pom which is generally validated)
 - these html .module files were then cached and returned to gradle
 - large numbers of gradle builds started failing

So in summary i'd be less worried about total reproducibility (which is in essence an illusion) and more about the risk to large numbers of projects of incorrect repository content in absence of repository validation.

**Comment by cobexer** [comment]
Thank you for your interest in Gradle!

This is the intended behavior, and there’s no plan to change it. This issue will be closed.
***
Hiding the problem from you (by only issuing a warning) is not a proper solution to external services with poisoned caches.
